pkg/azuredisk/controllerserver.go

```diff
@@ -927,6 +983,10 @@ func (d *Driver) DeleteSnapshot(ctx context.Context, req *csi.DeleteSnapshotRequ
 		}
 	}
 
+	if hasLock := d.volumeLocks.HasLock(snapshotName); hasLock {
+		return nil, status.Errorf(codes.Aborted, "snapshot %s is being copied to a new region", snapshotName)
```

Comment:
this is incorrect, just use following way to lock:
```
	if acquired := d.volumeLocks.TryAcquire(snapshotName); !acquired {
		return nil, status.Errorf(codes.Aborted, volumeOperationAlreadyExistsFmt, snapshotName)
	}
	defer d.volumeLocks.Release(snapshotName)
```

Reply:
done. No need to add lock in delete snapshot.